compare_mt.py

```diff
@@ -30,36 +31,26 @@ def print_score_report(ref, out1, out2,
     score_type: A string specifying the scoring type (bleu/length)
   """
   scorer = scorers.create_scorer_from_profile(score_type)
-  print(f'{scorer.name()}:')
   score1, str1 = scorer.score_corpus(ref,out1)
   score2, str2 = scorer.score_corpus(ref,out2)
-  if str1 is not None:
-    print(f' Sys1: {score1} ({str1})\n Sys2: {score2} ({str2})')
-  else:
-    print(f' Sys1: {score1}\n Sys2: {score2}')
+  stats = {'scorer_name': scorer.name(), 'score1': score1, 'str1': str1, 'score2': score2, 'str2': str2}
```

Comment:
The way of doing things here is reasonable, but there is one caveat: just using a dictionary might be too flexible. An alternative would be to have a constructor for `ScoreReport()` that takes all of the arguments that we expect the score report to have. This would make it easier to see which stats the score report expects, and perhaps reduce bugs.

Does this make sense? I'm not sure if I'm explaining this clearly, so if you don't understand I can do an example of this myself.

Reply:
This makes sense to me. I can create a constructor for each subclass of Report. That would make the subclass more readable. I will modify this.